### ISSUE TITLE
isPurchased with subscriptions

### ISSUE THREAD

**Issue by jofuelo** [russell-archer/StoreHelper#6]
Hi, 

I wanted to ask/suggest what happens to the function isPurchased when dealing with a subscription that may be cancelled or/and expired. The desirable behaviour is that isPurchased would return true when the subscription is autorenewing normally, or when it has been cancelled but still hasn't reached it's expiration date. And false otherwise, that is to say, when the subscribed content should be no longer available to the user. I haven't seen any status confirmation in the function, so I assume it returns true if it has been purchased at any time, even if it has been cancelled and expired.

Am I correct? If I am, what would be an optimal way to check this circumstance? The idea is to have a function that returns true if the user should have access to that premium content and false otherwise.

Thank you.

**Comment by russell-archer** [comment]
Hi,

Sorry for the delay in getting back to you. I just ran a few tests to check that StoreHelper behaves in the way you suggest, which I agree is the correct behaviour.

Using the StoreHelperDemo app I did the following (I set the renewal rate in Xcode to be 5 minutes):

1. With no products subscribed to
isPurchased = false for all subscription products

2. Subscribed to the "Silver" product
isPurchased = true for the "Silver" product

3. Cancelled the subscription to the "Silver" product
isPurchased = true for the "Silver" product

4. Repeated step 3 while the subscription was still valid (before it expired)
isPurchased = true for the "Silver" product

5. When the subscription for the "Silver" product had expired
isPurchased = false for the "Silver" product

So, it looks like everything is working as expected from my (limited) testing.

Hope this helps. If anything's unclear let me know!
